kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/WorkflowViewStub.kt

```diff
@@ -0,0 +1,62 @@
+package com.squareup.workflow.ui
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.View
+import android.view.ViewGroup
+
+/**
+ * A placeholder [View] that can replace itself with ones driven by workflow renderings,
+ * similar to [android.view.ViewStub].
+ */
+class WorkflowViewStub(
+  context: Context,
+  attributeSet: AttributeSet? = null
+) : View(context, attributeSet) {
+  init {
+    setWillNotDraw(true)
+  }
+
+  /**
+   * By starting as our own delegate, we make the parent checking code in [update] simpler.
+   */
+  private var _actual: View = this
+
+  /**
+   * On-demand access to the delegate established by the last call to [update],
+   * or null if none has been set yet.
+   */
+  val actual: View? get() = _actual.takeIf { it !is WorkflowViewStub }
+
+  /**
+   * Replaces this view with one that can display [rendering]. If it
+   * has already been replaced, updates that replacement if it [canShowRendering].
+   * If it can't, a new replacement is put in place.
+   *
+   * @return the view that showed [rendering]
+   * @throws RuntimeException if [rendering] cannot be displayed, typically indicating
```

Comment:
Where does this exception actually get thrown? I don't see it. If it actually is thrown, could we throw something more specific, like `IllegalStateException` or `AssertionError`?

Reply:
It's thrown by ViewRegistry, and could be either `IllegalStateException` or `IllegalArgumentException`. See [ViewRegistry.buildView], line 90. I thought about changing the `check` call there to `require`, but it just seemed wrong. And wrapping and re-throwing here to clean up the kdoc seems silly.

Reply:
Document those two exceptions directly then? Don't feel strongly though.

Reply:
Saw this after merging. I'll figure out something, agree this is weird.